exchange/seat_non_bids_test.go

```diff
@@ -0,0 +1,109 @@
+package exchange
+
+import (
+	"testing"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/exchange/entities"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestSeatNonBidsAdd(t *testing.T) {
+	type fields struct {
+		seatNonBidsMap map[string][]openrtb_ext.NonBid
+	}
+	type args struct {
+		bid          *entities.PbsOrtbBid
+		nonBidReason int
+		seat         string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+		want   map[string][]openrtb_ext.NonBid
+	}{
+		{
+			name:   "nil seatNonBidsMap",
+			fields: fields{seatNonBidsMap: nil},
+			args:   args{},
+			want:   nil,
+		},
+		{
+			name:   "nil seatNonBidsMap with bid object",
+			fields: fields{seatNonBidsMap: nil},
+			args:   args{bid: &entities.PbsOrtbBid{Bid: &openrtb2.Bid{}}, seat: "bidder1"},
+			want:   sampleSeatNonBidMap("bidder1", 1),
+		},
+		{
+			name:   "multiple nonbids for same seat",
+			fields: fields{seatNonBidsMap: sampleSeatNonBidMap("bidder2", 1)},
+			args:   args{bid: &entities.PbsOrtbBid{Bid: &openrtb2.Bid{}}, seat: "bidder2"},
+			want:   sampleSeatNonBidMap("bidder2", 2),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			snb := &nonBids{
+				seatNonBidsMap: tt.fields.seatNonBidsMap,
+			}
+			snb.addBid(tt.args.bid, tt.args.nonBidReason, tt.args.seat)
+			assert.Equalf(t, tt.want, snb.seatNonBidsMap, "expected seatNonBidsMap not nil")
+		})
+	}
+}
+
+func TestSeatNonBidsGet(t *testing.T) {
```

Comment:
Could you add an extra test case here where seatNonBids is equal to nil

Reply:
Done